apps/prairielearn/src/pages/shared/syncHelpers.js

```diff
@@ -121,96 +133,79 @@ function logProgressOutput(output, job, printedInfos, prefix) {
  *
  * @param {string} image
  * @param {import('@prairielearn/docker-utils').DockerAuth} dockerAuth
- * @param {Pick<import('../../lib/server-jobs').ServerJob, 'info'>} job
- * @param {Function} callback
+ * @param {import('../../lib/server-jobs').ServerJob} job
  */
-function pullAndPushToECR(image, dockerAuth, job, callback) {
+async function pullAndPushToECR(image, dockerAuth, job) {
   const { cacheImageRegistry } = config;
   if (!cacheImageRegistry) {
-    return callback(new Error('cacheImageRegistry not defined'));
+    throw new Error('cacheImageRegistry not defined');
   }
 
   const repository = new DockerName(image);
-  const params = {
+  job.info(`Pulling ${repository.getCombined()}`);
+  const pullStream = await docker.createImage({
     fromImage: repository.getRepository(),
     tag: repository.getTag() || 'latest',
-  };
-  job.info(`Pulling ${repository.getCombined()}`);
-  docker.createImage({}, params, (err, stream) => {
-    if (ERR(err, callback)) return;
-    if (!stream) throw new Error('Missing stream from createImage()');
```

Comment:
Is this check not necessary anymore?

Reply:
IIRC this was only necessary to make TypeScript happy with the errback here. If `createImage` does not throw, it is guaranteed to return a stream.